scipy/stats/_hypotests.py

```diff
@@ -130,3 +131,133 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     p = chi2.sf(w, r)
 
     return Epps_Singleton_2sampResult(w, p)
+
+
+def poisson_etest(k1, k2, n1, n2, diff=0, alternative='two-sided'):
+    """
+    Calculate E-test for the mean difference of two samples
+
+    The test requires two samples coming from poisson distribution. This is a
+    right-sided and two-sided test.
+
+    Parameters
+    ----------
+    k1, k2 : int or float
+        Count from the first and second samples respectively
+    n1, n2 : int or float
+        Sample size of each sample
+    diff : int of float, optional
+        The difference of mean between two samples under null hypothesis
+    alternative : {'two-sided', 'greater'}, optional
```

Comment:
what about `less`

Reply:
The paper does not specify any hypothesis about `less`. It is possible to add that by switching the arguments. But do you think that by adding `less` might miss-interpret how e-test works?

Reply:
Switching the arguments has the downside of changing the test statistic based on the `alternative`. Perhaps this is a convention for some tests, but it's not clear that it's the case here, so I hope we can keep the test statistic value the same regardless of alternative.